quic/s2n-quic-core/src/xdp/encoder.rs

```diff
@@ -141,11 +141,7 @@ fn encode_ipv4<M: Message<Handle = path::Tuple>>(
         *header.destination_mut() = remote_ip;
 
         // calculate the IPv4 header checksum
-        {
-            let mut checksum = state.cached_checksum;
-            checksum.write(header.as_bytes());
-            header.checksum_mut().set(checksum.finish());
-        }
+        header.update_checksum();
```

Comment:
is the `cached_checksum` still updated somewhere?

Reply:
No it was just to reduce the amount of probing performed

https://github.com/aws/s2n-quic/blob/eae3b081d29f1f3caae02eb65b4d55af0b7d57c5/quic/s2n-quic-core/src/xdp/encoder.rs#L30-L31

Reply:
Its ok that its still returned here though? https://github.com/aws/s2n-quic/pull/1846/files#diff-2c299aa60c3e5f0f4c418d64cef7ec16cbac8d37c769084721570b235b6e4f14R61

Reply:
Yeah the checksum implements `Copy` so it's being copied there.

Reply:
I see, the probing is happening in the `impl Default for Checksum`